api/v1alpha1/terraform_types.go

```diff
@@ -85,6 +85,9 @@ type TerraformSpec struct {
 	// +optional
 	BackendConfigsFrom []BackendConfigsReference `json:"backendConfigsFrom,omitempty"`
 
+	// +optional
+	Workspace string `json:"workspace,omitempty"`
```

Comment:
I feel like `Workspace` should be part of `BackendConfig`

Reply:
Isn't BackendConfig directly mapped into backend {} terraform's config?

Reply:
I see now, I agree it should be part of BackendConfigSpec

Reply:
That's true. I think it can be in `BackendConfig` since it is really a feature supported by a Terraform `backend`. However, it is also a top level command line option of `terraform`. I can go either way on this.

Reply:
Moved to BackendConfig